---
title: "Whooping Crane data analysis: Fecal Samples"
author: "Jess Diaz"
date: "`r format(Sys.time(), '%a %d %b')`"
output:
  rmdformats::readthedown
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Setup
```{r load packages, include=FALSE}
library(qiime2R)
library(microbiome)
library(dplyr)
library(ggplot2)
library(vegan)
library(decontam)
library(nlme)
library(rmdformats)
library(microbiomer)
library(car)
library(visreg)
library(emmeans)
library(pairwiseAdonis)
source("HighstatLibV10.R")
library(ANCOMBC)
library(tibble)
```

```{r set global parameters, include=FALSE}
theme_set(theme_classic())

sc <- scale_color_manual(values = c("AR" = "#C34554", 
                                    "AR-FMT" = "#55B35E", 
                                    "PR" = "#81B1F7")) 
```

```{r functions, include=FALSE}
# calculates values for stats bars for plots
data.summary <- function(x) {
  m <- mean(x)
  ymin <- m-sd(x)
  ymax <- m+sd(x)
  return(c(y=m, ymin=ymin, ymax=ymax))
}
```

```{r create phyloseq object, include = FALSE}
# create master phyloseq
WC <-qza_to_phyloseq(
  features="table-filt-bytaxa.qza",
  tree="rooted-tree.qza",
  taxonomy="taxonomy.qza",
  metadata = "fecal_qiime_metadata_corrected.txt"
)

# sum contrasts for categorical variables
sample_data(WC)$day <- C(factor(sample_data(WC)$day), sum)
sample_data(WC)$year <- C(factor(sample_data(WC)$year), sum)
sample_data(WC)$group <- C(factor(sample_data(WC)$group), sum)
sample_data(WC)$bird.code <- C(factor(sample_data(WC)$bird.code), sum)
sample_data(WC)$sex <- C(factor(sample_data(WC)$sex), sum)

# remove blank 5 (not from my study)
WC <- subset_samples(WC, bird.code != "Blank-5")
```

This document is intended for initial analysis of the Whooping Crane fecal sequencing data. Included in this analysis are 122 samples spread across 34 chicks, 5 years, and 5 timepoints. 2 extraction blanks were included in this analysis, as well as several PCR blanks which are not included here (Jose is making sure that they look clean).  

# Data Cleanup and Setup

## Data Summary
```{r data summaries, echo = FALSE, warnings = FALSE, message = FALSE}
# data summaries
summarize_phyloseq(WC)
```
## Decontam

First I will use the extraction blanks to identify contaminants. Using a threshold of 0.5, these are the taxa identified as contaminants:

```{r decontam 0.5, echo = FALSE}
# make new variable where blanks are "TRUE"
sample_data(WC)$is.neg <- sample_data(WC)$sample.type == "blank"

# run decontam
contamdf.prevalence <- isContaminant(WC, method="prevalence", neg="is.neg", threshold=0.5)

# identify which taxa are contaminants
table(contamdf.prevalence$contaminant)
contams <- which(contamdf.prevalence$contaminant)

# list contaminants
tax <- as(tax_table(WC), "matrix")
tax[contams,]
```
I will also make a plot that shows prevalence in the negatives and in true samples, to check how they are getting classified.  
```{r decontam plot, echo = FALSE}
# presence-absence phyloseq for controls and samples
ps.pa <- transform_sample_counts(WC, function(abund) 1*(abund>0))
ps.pa.neg <- prune_samples(sample_data(ps.pa)$sample.type == "blank", ps.pa)
ps.pa.pos <- prune_samples(sample_data(ps.pa)$sample.type == "fecal", ps.pa)

# make dataframe for prevalence
df.pa <- data.frame(pa.pos=taxa_sums(ps.pa.pos), pa.neg=taxa_sums(ps.pa.neg),
                    contaminant=contamdf.prevalence$contaminant)
ggplot(data=df.pa, aes(x=pa.neg, y=pa.pos, color=contaminant)) +
  geom_point() +
  xlab("Prevalence (blanks)") +
  ylab("Prevalence (samples)") +
  ggtitle("Threshold = 0.5") +
  geom_abline(slope = 1, intercept = 0)
```

I am not happy with how this identified contaminants. There are some that are still present in a very large number of samples and were only found in one contaminant, and one that was found in a contaminant but not many samples and was not identified as a contaminant. I am going to try lowering the strict threshold of 0.5 to 0.1 to see if that keeps some of these samples.  
```{r decontam 0.1, echo = FALSE}
# run decontam
contamdf.prevalence <- isContaminant(WC, method="prevalence", neg="is.neg", threshold=0.1)

# identify which taxa are contaminants
table(contamdf.prevalence$contaminant)
contams <- which(contamdf.prevalence$contaminant)

# list contaminants
tax <- as(tax_table(WC), "matrix")
tax[contams,]

# presence-absence phyloseq for controls and samples
ps.pa <- transform_sample_counts(WC, function(abund) 1*(abund>0))
ps.pa.neg <- prune_samples(sample_data(ps.pa)$sample.type == "blank", ps.pa)
ps.pa.pos <- prune_samples(sample_data(ps.pa)$sample.type == "fecal", ps.pa)

# make dataframe for prevalence
df.pa <- data.frame(pa.pos=taxa_sums(ps.pa.pos), pa.neg=taxa_sums(ps.pa.neg),
                    contaminant=contamdf.prevalence$contaminant)
ggplot(data=df.pa, aes(x=pa.neg, y=pa.pos, color=contaminant)) +
  geom_point() +
  xlab("Prevalence (blanks)") +
  ylab("Prevalence (samples)") +
  ggtitle("Threshold = 0.1") +
  geom_abline(slope = 1, intercept = 0)
```
These make a bit more sense as contaminants. Now I just want to make sure that I am not removing too many samples by removing these contaminants. This table shows % of reads removed from that sample after removing contaminants.  
```{r remove contaminants, echo = FALSE}
# remove taxa from phyloseq
WC.decontam <- prune_taxa(!contamdf.prevalence$contaminant, WC)

# set up df with readcounts and % removed
readcounts <- data.frame(matrix(ncol = 0, nrow = 124))
readcounts$pre <- as.data.frame(sample_sums(WC))
readcounts$post <- sample_sums(WC.decontam)
readcounts$per.removed <- ((readcounts$pre - readcounts$post)/readcounts$pre) *100

print(readcounts$per.removed)

# print otu table for contaminants
as.data.frame(otu_table(WC)) %>% slice(contams)
```
These look ok to me. Though some of them have higher read counts than in the blanks, they are only in a few samples, and the blanks had much lower reads overall so that would skew the number. Something else I could do would be to compare which blanks were extracted with which samples, which would give me a better idea if they may be true extraction contaminants or not.  

6 taxa were identified as contaminants using a threshold of 0.1 (would be a good idea for me to articulate what this threshold means). I have removed these taxa from analysis.

## Rarefaction

Now I can look at the read counts to see how many reads are left in each sample.
```{r read counts, echo = FALSE}
sample_sums(WC.decontam)
```
Based on the read counts, I will rarefy at 8113 to retain all data points except the three that had below 1000 reads (hopefully will get resequenced). Blanks should be automatically removed with the rarefaction.

```{r rarefy, include=FALSE}
WC.rare <- rarefy_even_depth(WC.decontam, sample.size=8113, rngseed=14, verbose=TRUE) # rarefy
```

# Alpha diversity

## Initial plots

After rarefying, I can look at the initial alpha diversity plots to look for differences.

```{r alpha all, echo = FALSE}
plot_richness(WC.rare, x="group")
plot_richness(WC.rare, x="year", color = "group")
plot_richness(WC.rare, x="sex")
plot_richness(WC.rare, x="day", color="group")
```

Some trends I see: PR might have slightly higher diversity than other groups, and the green AR-FMT points seem to be a bit lower. Something is definitely going on with day 14.  

Digging into differences between groups, I can plot diversity per group over time. I removed day 21 for these plots because only one group has any data points at day 21.

```{r alpha dataframe setup, include = FALSE}
# save alpha diversity calculations as alpha dataframe
alpha <- microbiome::alpha(WC.rare)

# save phyloseq metadata as separate object, and add shannon and observed richness
meta.rare <- meta(WC.rare) %>%
  mutate(shannon = alpha$diversity_shannon) %>%
  mutate(observed = alpha$observed)
```

Shannon diversity plot:  
```{r shannon over time, echo = FALSE}
meta.rare %>%
  filter(day != "21") %>% # remove day 21 samples because only AR had it
  ggplot(aes(x=day, y=shannon, group = group)) + sc +
  geom_point(aes(color = group), position = position_dodge(width = 0.1), shape = 1, size = 3) +
  stat_summary(fun=mean, geom = "line", linewidth = 1, aes(color = group)) +
  stat_summary(fun.data = data.summary, geom = "errorbar", linewidth = 1, width = 0.5, na.rm = TRUE, aes(color = group)) +
  stat_summary(fun=mean, geom = "point", size = 3, aes(color = group)) +
  theme_classic(base_size = 15) +
  ylab("shannon diversity")
```

Observed richness plot:  
```{r richness over time, echo = FALSE}
meta.rare %>%
  filter(day != "21") %>%
  ggplot(aes(x=day, y=observed, group = group)) + sc +
  geom_point(aes(color = group), position = position_dodge(width = 0.1), shape = 1, size = 3) +
  stat_summary(fun=mean, geom = "line", linewidth = 1, aes(color = group)) +
  stat_summary(fun.data = data.summary, geom = "errorbar", linewidth = 1, width = 0.5, na.rm = TRUE, aes(color = group)) +
  stat_summary(fun=mean, geom = "point", size = 3, aes(color = group)) +
  theme_classic(base_size = 15) +
  ylab("observed richness")
```

Seems like potentially, richness converges over time between the three groups.  

```{r extra code for richness over time, include = FALSE}
# these plots were generated for my rotation talk, to add in data sequentially

# observed richness but no AR-FMT group
# meta.rare.noblank %>% 
#   filter(day != "21") %>% 
#   ggplot(aes(x=day, y=observed, group = group, alpha = group)) + sc +
#   scale_alpha_manual(values = c(1, 0, 1)) +
#   geom_point(aes(color = group), position = position_dodge(width = 2), shape = 1, size = 3) +
#   stat_summary(fun=mean, geom = "line", linewidth = 1, aes(color = group)) +
#   stat_summary(fun.data = data.summary, geom = "errorbar", linewidth = 1, width = 2, na.rm = TRUE, aes(color = group)) +
#   stat_summary(fun=mean, geom = "point", size = 3, aes(color = group)) +
#   theme_classic(base_size = 15) +
#   ylab("observed richness")
# 
# # observed richness but no AR-or AR groups
# meta.rare.noblank %>% 
#   filter(day != "21") %>% 
#   ggplot(aes(x=day, y=observed, group = group, alpha = group)) + sc +
#   scale_alpha_manual(values = c(0, 0, 1)) +
#   geom_point(aes(color = group), position = position_dodge(width = 2), shape = 1, size = 3) +
#   stat_summary(fun=mean, geom = "line", linewidth = 1, aes(color = group)) +
#   stat_summary(fun.data = data.summary, geom = "errorbar", linewidth = 1, width = 2, na.rm = TRUE, aes(color = group)) +
#   stat_summary(fun=mean, geom = "point", size = 3, aes(color = group)) +
#   theme_classic(base_size = 15) +
#   ylab("observed richness")
```

## Statistics for ZYMO
This analysis was for the ZYMO grant. I compared richness between AR and PR chicks at 14 days to detect whether there was a statistical difference (there was). It does throw up an error about "ties", not sure what that means but based on my googling hopefully it wouldn't affect the overall conclusion of significance here. I did this using two different packages (slightly different tests?) but they gave the same results.  
```{r Wilcox-Mann-Whitney for ZYMO grant, echo = FALSE}
# filter table to isolate day 14 AR and PR
meta.rare.d14.noFMT <- meta.rare %>%
  filter(day == "14") %>%
  filter(group != "AR-FMT") %>%
  mutate(day = as.factor(day)) %>%
  mutate(group = as.factor(group))

# compare means
wilcox.test(observed ~ group, data=meta.rare.d14.noFMT)
coin::wilcox_test(observed ~ group, data=meta.rare.d14.noFMT)
```
## Shannon Linear Model
I am constructing a model to detect the factors influencing alpha diversity in these Whooping Crane samples. The potential explanatory factors I have are: Individual bird, year, sex, day, and group.  

*According to Rachel, I might want to consider making this a GAM.*

First I will make a pairs plot.  
```{r pairs, warning = FALSE, echo = FALSE, message = FALSE}
# Make pairs plot
meta.forpairs <- meta.rare %>%
  select(-bird.name, -sample.type, -date.collected, -date.extracted, -sample.weight, -extraction.id, -ng.uL, -A260.280, -is.neg) %>%
  filter(day != "21")
pairs(meta.forpairs, lower.panel=panel.smooth, diag.panel=panel.hist,
upper.panel=panel.cor)
```

The shannon plot looks more even than observed richness so I will use that one for analysis. I will not look for collinearity since all of my explanatory variables are categorical.

I will now do a linear mixed model with *shannon* as my response variable, *day*, *group*, *year* and *sex* as my explanatory variables, and *bird.code* as my random variable to correct for differences between birds. I will start with no interaction terms. Here is the full equation for the base model.
$$
i = bird\ ID
$$
$$
j = individual\ sample
$$

**Stochastic part:**
$$
Richness_{ij} \sim N(Richness_{det\ ij}, \sigma^2)
$$
**Deterministic part:**
$$
Richness_{det\ ij} = \alpha + \beta_1 (day_{ij}) + \beta_2 (group_{ij}) + \beta_3 (sex_{ij}) + \beta_4 (year_{ij}) + a_i
$$
**Stochastic part for random effect:**
$$
a_i \sim N(0, \sigma^2_a)
$$
Now I will fit the full model, removing day 21 from analysis. I need to fit using maximum likelihood so that I can compare the later models which have different fixed effects. I am using the *nlme* package because the models will not converge (error: boundary (singular) fit: see help('isSingular')) when I try using *lme4*. This means I cannot use the dHARMA package for looking at residuals.  
```{r fit model, echo = FALSE}
meta.no21 <- meta.rare %>% 
  filter(day != "21")

model <- lme(shannon ~ day + group + sex + year, random=~1|bird.code, 
               method = "ML", data = meta.no21)
```

I will also compare the AIC with other versions of the model as listed below. For now, I am sticking with a general linear model as the histogram of shannon index looks pretty normally distributed.  

  1. day | bird  
  2. group + day | bird  
  3. group * day | bird  
  4. group * day + sex | bird  
  5. group * day + year | bird  
  6. group * day + sex + year | bird  
  
Below are the AIC values for each model.
```{r fit alternative models, echo = FALSE}
model1 <- lme(shannon ~ day, random=~1|bird.code, 
             method = "ML", data = meta.no21)
model2 <- lme(shannon ~ day + group, random=~1|bird.code, 
             method = "ML", data = meta.no21)
model3 <- lme(shannon ~ day * group, random=~1|bird.code, 
             method = "ML", data = meta.no21)
model4 <- lme(shannon ~ day * group + sex, random=~1|bird.code, 
               method = "ML", data = meta.no21)
model5 <- lme(shannon ~ day * group + year, random=~1|bird.code, 
               method = "ML", data = meta.no21)
model6 <- lme(shannon ~ day * group + sex + year, random=~1|bird.code, 
               method = "ML", data = meta.no21)

AIC(model)
AIC(model1)
AIC(model2)
AIC(model3)
AIC(model4)
AIC(model5)
AIC(model6)
```
Out of all the models, the most supported is the full model that includes an interaction term between group and day. I will refit the model using REML and use that to look at statistical conclusions about the data going forward, first checking the model fit. 

```{r model validation, echo = FALSE}
model6 <- lme(shannon ~ day * group + sex + year, random=~1|bird.code, 
               method = "REML", data = meta.no21)

# plot residuals against fitted values
plot(model6)

# plot residuals against explanatory variables
plot(model6$residuals ~ model6$data$day)
plot(model6$residuals ~ model6$data$group)
plot(model6$residuals ~ model6$data$year)
plot(model6$residuals ~ model6$data$sex)

# Q-Q plot to check for normality
qqnorm(resid(model6), pch=16)
qqline(resid(model6))
```

The model passes. Now some results:  

```{r model results, echo = FALSE, warnings = FALSE, message = FALSE}
summary(model6)

# test significance of terms
car::Anova(model6)
# drop1(model6, test='Chisq')
```
Based on the model test using Anova(), adding sex and year to the model doesn't seem to change it's fit very much. But adding the interaction term between day and group makes a huge difference, significantly improving the model. I couldn't use drop1() for this because it didn't give an AIC?  

Using visreg(), I can visualize the model. Might be worth making year and/or day a continuous variable?
```{r visualize model, echo = FALSE}
visreg(model6, xvar = "day", by = "group", data = meta.no21)
```
## Post-hoc tests

Now I will do post-hoc tests between all my combinations of groups to see what is statistically different from each other. I am using a multiple comparison with Tukey adjustment through the methods described here: https://biostats.w.uib.no/post-hoc-tests-multiple-comparisons-in-linear-models-2/  

```{r tukey, echo = FALSE}
# gives means for each group/day pair
emm <- emmeans(model6, ~ group * day)

# give p-value comparisons for each pair
pairs(emm, simple = "each", adjust = "holm")
```

# Beta diversity

## Initial plots

I will do some initial beta diversity plots to look for clustering patterns.

```{r bray curtis setup, include = FALSE}
# no day 21
WC.rare.no21 <- subset_samples(WC.rare, day != "21") # remove day 21 from phyloseq
BrayDist.no21 <- distance(WC.rare.no21, method = "bray") # create bray dist matrix
BrayOrd.no21 <- ordinate(WC.rare.no21, "NMDS", distance = BrayDist.no21) # make NMDS ordination
BrayPlot.no21 <- plot_ordination(WC.rare.no21, BrayOrd.no21, justDF = TRUE) # make base NMDS plot
```

Bray Curtis by group:  
```{r bray by group, echo = FALSE}
# just group
BrayPlot.no21 %>%
  ggplot(aes(x=NMDS1, y=NMDS2)) + sc +
  geom_point(size=4, aes(color = group)) +
  labs(shape = "day")
```

Adding day as shape:  
```{r bray by group and day, echo = FALSE}
# group and day
BrayPlot.no21 %>%
  ggplot(aes(x=NMDS1, y=NMDS2)) + sc +
  geom_point(size=4, aes(color = group, shape = day)) +
  labs(shape = "day")
```

I get the sense here too that beta diversity seems to be converging over time. How can I test this?

```{r extra code for bray by group, include = FALSE}
# these plots were generated for my rotation talk, to add in data sequentially

# by group but only PR and AR groups
# BrayPlot.noblank.no21 %>% 
#   ggplot(aes(x=NMDS1, y=NMDS2, alpha = group)) + sc +
#   scale_alpha_manual(values = c(1, 0, 1)) +
#   geom_point(size=4, aes(color = group)) +
#   labs(shape = "day")
# 
# # by group but only PR groups
# BrayPlot.noblank.no21 %>% 
#   ggplot(aes(x=NMDS1, y=NMDS2, alpha = group)) + sc +
#   scale_alpha_manual(values = c(0, 0, 1)) +
#   geom_point(size=4, aes(color = group)) +
#   labs(shape = "day")
```


## PERMANOVA

First I want to see if PERMANOVA identifies whether the composition is significant using the same model structure found for alpha diversity (group * day + sex + year), and including bird.code as a fixed effect because adonis2 cannot do random effects. I plan to move this to Primer7 eventually which should be able to handle random effects.  
```{r adonis, echo = FALSE}
adonis2(BrayDist.no21 ~ group * day + sex + year + bird.code, data = meta(WC.rare.no21))
```
All terms are significant except for bird.code. 

Next step will be post-hoc tests between each of the groups. For this I am using the pairwiseAdonis package.
```{r pairwise Adonis, echo = FALSE}
# pairwise by group
pairwise.adonis(BrayDist.no21, meta(WC.rare.no21)$group)

# pairwise by day
pairwise.adonis(BrayDist.no21, meta(WC.rare.no21)$day)

# pairwise by year
pairwise.adonis(BrayDist.no21, meta(WC.rare.no21)$year)
```
Looks like PR is different from both AR and AR-FMT, but those are not different to each other. The day pairwise comparison confirms movement over time, where the early timepoints are different from the later timepoints. Additionally, all years were significantly different from each other so year should definitely continue to be accounted for in the analyses.  

Now is there a way to dig into the direction of movement over time?

## Beta dispersion analysis

Now I will conduct a beta dispersion analysis to see if the groups differ in their variance.  
```{r beta disper, echo = FALSE}
disp <- vegan::betadisper(BrayDist.no21, meta.no21$group)
anova(disp)
TukeyHSD(disp)
```
It seems that the AR and AR-FMT groups are not different in variance, but PR and AR are super different. PR and AR-FMT are almost different. This lines up with how the bray curtis graphs look.

## Differential Abundance (ANCOM-BC)

```{r Old ANCOM code, include = FALSE}
# plot.ancombc.asc <- function(phy_object, taxonomy, grouping) {
#   tax_data = aggregate_taxa(phy_object, taxonomy)
#   tax_mat = as(tax_table(tax_data), "matrix")
#   ancombc_out = ancombc(phyloseq = tax_data, formula = grouping, p_adj_method = "BH", group = grouping)
#   res = ancombc_out$res
#   res_global = ancombc_out$res_global
# 
#   df_fig1 = data.frame(res$beta * res$diff_abn, check.names = FALSE) %>%
#     rownames_to_column("ASV")
#   colnames(df_fig1)[-1] = "BETA"
#   df_fig2 = data.frame(res$se * res$diff_abn, check.names = FALSE) %>%
#     rownames_to_column("ASV")
#   colnames(df_fig2)[-1] = "SD"
#   df_fig = df_fig1 %>% left_join(df_fig2, by = "ASV") %>%
#     transmute(ASV, BETA, SD) %>%
#     filter(BETA != 0) %>% arrange(BETA) %>%
#     mutate(group = ifelse(BETA >0, "g1", "g2"))
#   df_fig$ASV = factor(df_fig$ASV, levels = df_fig$ASV)
# 
#   return(df_fig)
# }

# # subset samples to remove AR-FMT group and isolate day 14
# WC.noFMT.d14 <- subset_samples(WC.rare.no21, group != "AR-FMT" & day == "14")
# 
# # run ANCOMBC based on order and zoo status
# set.seed(14)
# ANCOM.O.ARPR.d14 <- plot.ancombc.asc(WC.noFMT.d14, "Order", "group")
# # remove unknowns
# ANCOMOrdFUFR <- ANCOMOrdFUFR[ANCOMOrdFUFR$ASV != "Unknown",]
# 
# # make list of phyla represented by identified orders
# ANCOMOrdFUFRPhyla <- vector(mode = "list", length = 13)
# for(row in 1:13) {
#   order <- ANCOMOrdFUFR$ASV[row]
#   print(order)
#   for(tax in 1:length(t(ASV_taxonomy_decontam))) {
#     if(is.na(ASV_taxonomy_decontam$Order[tax])) {
#       print("NA")
#     } else if(ASV_taxonomy_decontam$Order[tax] == order){
#       phylum <- ASV_taxonomy_decontam$Phylum[tax]
#       print("match")
#       print(phylum)
#       ANCOMOrdFUFRPhyla[row] <- as.character(phylum)
#       break
#     } else {
#     }
#   }
# }
# ANCOMOrdFUFR$Phylum <- ANCOMOrdFUFRPhyla
# ANCOMOrdFUFR$Phylum <-factor(ANCOMOrdFUFR$Phylum, levels = c("Actinobacteriota", "Firmicutes", "Proteobacteria", "Desulfobacterota", "Deferribacterota"))
```
For differential abundance, I want to compare which taxa are more prevalent between the three treatments.
```{r ANCOMBC2 AR vs PR, message = FALSE, warnings = FALSE}
WC.d14 <- subset_samples(WC.rare.no21, day == "14")
# ancom2.test <- ancombc2(data = WC.noFMT.d14, tax_level = "Order", fix_formula = "group", rand_formula = "bird.code",
#   p_adj_method = "BH", group = "group")
```



# Campylobacter and Salmonella

## Initial exploration

I would like to compare the relative abundance of bacterial taxa of Campylobacter and Salmonella genuses between rearing groups, because these are typically screened in Whooping Cranes as potential pathogens (Keller and Hartup 2013). I expect there to be a high number of cranes with Campylobacter (at least among captive reared) and a low number with Salmonella.  

```{r Campylobacter, echo = FALSE}
# transform read counts to relative abundance
WC_relative <- microbiomer::to_RA(WC)

# merge taxa at the genus level
WC_familyglom <- tax_glom(WC_relative, taxrank = rank_names(WC)[5], NArm = FALSE)

# melt phyloseq object into dataframe
FamilyGlom <- psmelt(WC_familyglom)

# isolate campylobacter
FamilyGlom.Campy <- FamilyGlom %>% 
  filter(Family == "Campylobacteraceae") %>% 
  filter(sample.type == "fecal") %>% 
  filter(day != "21") %>% 
  mutate(Campy_YN = if_else(Abundance > 0, "Yes", "No"))

# plot
# FamilyGlom.Campy %>% 
#   ggplot(aes(x=group, y=Abundance)) + sc +
#   geom_point(size=4, aes(color = group, shape = day)) +
#   facet_wrap(vars(day))

FamilyGlom.Campy %>% 
  ggplot(aes(x=day, y=Abundance, group = group)) + sc +
  geom_point(aes(color = group), position = position_dodge(width = 0.1), shape = 1, size = 3) +
  stat_summary(fun=mean, geom = "line", linewidth = 1, aes(color = group)) +
  stat_summary(fun.data = data.summary, geom = "errorbar", linewidth = 1, width = 0.5, na.rm = TRUE, aes(color = group)) +
  stat_summary(fun=mean, geom = "point", size = 3, aes(color = group)) +
  theme_classic(base_size = 15) +
  ylab("Relative abundance of Campylobacter")

FamilyGlom.Campy %>% 
  ggplot(aes(x=day, fill = Campy_YN, color = Campy_YN)) +
  geom_bar(position = position_dodge2(), alpha = 0.75) +
  facet_wrap(vars(group))
```

```{r Campy table, include = FALSE}
ftable(xtabs(~ group+day+Campy_YN, data = FamilyGlom.Campy))
```

![Table made in excel. Shows percentage of Campylobacter-positive chicks in each group at each timepoint.](Campylobacter_table.png)

These plots and tables suggest that there is not much going on with the Campylobacter angle. All the groups and timepoints seem to have between 60-100% positive individuals with Campylobacter. This matches the 90% number in the Keller and Hartup paper from 2013 and might be interesting to note in an eventual paper.  

There was no Salmonella detected in any of the samples.

## Logistic regression (Campy)
*this section needs work*
I will use a similar model setup from the previous analyses to run a logistic regression testing whether presence of Campylobacter is influenced by group, day, and year, with bird ID as a random variable. *still need to put in bird ID*
```{r Logistic regression, echo = FALSE}
# turn YN into 0 and 1
FamilyGlom.Campy <- FamilyGlom.Campy %>%
  mutate(Campy_01 = if_else(Abundance > 0, 1, 0))

# fit model
campy_model <- glm(Campy_01 ~ group * day + year, family=binomial(link="logit"), data=FamilyGlom.Campy)

# plot model
FamilyGlom.Campy %>% 
  ggplot(aes(x=day, y = Campy_01, color = Campy_YN)) +
  geom_point(position = position_dodge2(width = 3), alpha = 0.75) +
  facet_wrap(vars(group))
plot(campy_model$fitted.values)

# check residuals
plot(residuals(campy_model, type="response") ~ campy_model$fitted.values)

simulationOutput <- simulateResiduals((fittedModel=campy_model))
plotResiduals(simulationOutput, quantiles=NULL)

# interpret results
summary(campy_model)
```



