---
title: "WC full analysis"
author: "Jess Diaz"
date: "`r format(Sys.time(), '%a %d %b')`"
output:
  rmdformats::readthedown
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Setup
```{r load packages, include=FALSE}
library(qiime2R)
library(microbiome)
library(dplyr)
library(ggplot2)
library(vegan)
library(decontam)
library(nlme)
library(rmdformats)
library(microbiomer)
library(car)
library(visreg)
library(emmeans)
library(pairwiseAdonis)
source("HighstatLibV10.R")
library(ANCOMBC)
library(tibble)
library(tidyr)
library(lemon)
library(MicEco)
```

```{r set global parameters, include=FALSE}
theme_set(theme_classic())

sc <- scale_color_manual(values = c("AR" = "#C34554", 
                                    "AR-FMT" = "#55B35E", 
                                    "PR" = "#81B1F7",
                                    "DI" = "gray")) 

knit_print.data.frame <- lemon_print
```

```{r functions, include=FALSE}
# calculates values for stats bars for plots
data.summary <- function(x) {
  m <- mean(x)
  ymin <- m-sd(x)
  ymax <- m+sd(x)
  return(c(y=m, ymin=ymin, ymax=ymax))
}
```

This document is intended to show the full analysis of the Whooping Crane data. Included in this analysis are:  
* 122 fecal samples spread across 34 chicks, 5 years, and 5 timepoints  
* ? tracheal samples spread across ? chicks, ? years, and 2 timepoints  

These results were sequenced over three runs:  
1) Run 1 included the fecal sample 16S sequencing plus 2 extraction blanks  
2) Run 2 included the tracheal sample 16S sequencing, the donor and inoculum fecal samples, and 3 extraction blanks  
3) Run 3 included the tracheal sample ITS sequencing (+ any blanks?)  

There were also several PCR blanks done with these runs at the sequencing facility. I never see those sequences but I believe Jose checks them to make sure they look clean.

# Data Import and Summary

## Import 16S data

I imported the 16S files from QIIME for runs 1 and 2 together. Note, I had to change the metadata file from what was used in QIIME. There were some "empty" rows and columns that were causing issues importing the data, so I just deleted those. I also added the "blank" sample type to the Run 2 blanks which were missing that.  

```{r create 16S phyloseq object, include = FALSE, message = FALSE}
# create master phyloseq
WC <- qza_to_phyloseq(
  features = "table-filt-bytaxa.qza",
  tree = "rooted-tree.qza",
  taxonomy = "taxonomy.qza",
  metadata = "WC_qiime_metadata_import2.txt"
)

# factor variables with some contrasts
sample_data(WC)$bird.code <- C(factor(sample_data(WC)$bird.code), sum)
sample_data(WC)$year <- C(factor(sample_data(WC)$year), sum)
sample_data(WC)$sex <- C(factor(sample_data(WC)$sex), sum)
sample_data(WC)$group <- C(factor(sample_data(WC)$group), sum)
sample_data(WC)$day <- C(factor(sample_data(WC)$day), sum)
sample_data(WC)$seq.run <- C(factor(sample_data(WC)$seq.run), sum)
sample_data(WC)$thawed <- C(factor(sample_data(WC)$thawed), sum)
sample_data(WC)$swab.type <- C(factor(sample_data(WC)$swab.type), sum)
```

## Decontam 16S data

I processed runs 1 and 2 through QIIME together so that they would have the same ASVs picked. However, for decontam I will separate them into their prospective runs, decontam separately according to the blanks that were associated with each run, then rearrange the data so that I have a group of fecal samples + donor and inoculum and a group of tracheal samples.  

```{r split phyloseq into runs, include = FALSE, message = FALSE}
WC1 <- subset_samples(WC, seq.run == "1")
WC2 <- subset_samples(WC, seq.run == "2")
```

### Run 1 decontam

First I will use the extraction blanks to identify contaminants. Using a threshold of 0.5, these are the taxa identified as contaminants:  

```{r decontam R1 0.5, echo = FALSE, message = FALSE}
# make new variable where blanks are "TRUE"
sample_data(WC1)$is.neg <- sample_data(WC1)$sample.type == "blank"

# run decontam
contamdf.prevalence <- isContaminant(WC1, method="prevalence", neg="is.neg", threshold=0.5)

# identify which taxa are contaminants
contams <- which(contamdf.prevalence$contaminant)

# list contaminants
tax <- as(tax_table(WC1), "matrix")
```

```{r R1 contam table 0.5, echo = FALSE, render = lemon_print}
# print number of contamninants and what they are
as.data.frame(table(contamdf.prevalence$contaminant))
as.data.frame(tax[contams,])
```

I will also make a plot that shows prevalence of taxa in the negatives and in true samples, to check how they are getting classified.  

```{r R1 decontam plot 0.5, echo = FALSE, message = FALSE}
# make separate presence-absence phyloseq for controls and samples
ps.pa <- transform_sample_counts(WC1, function(abund) 1*(abund>0))
ps.pa.neg <- prune_samples(sample_data(ps.pa)$sample.type == "blank", ps.pa)
ps.pa.pos <- prune_samples(sample_data(ps.pa)$sample.type == "fecal", ps.pa)

# make dataframe for prevalence in controls and samples, plus whether it was identified as a contaminant
df.pa <- data.frame(pa.pos=taxa_sums(ps.pa.pos), pa.neg=taxa_sums(ps.pa.neg),
                    contaminant=contamdf.prevalence$contaminant)

# plot prevalence and contaminant identification
ggplot(data=df.pa, aes(x=pa.neg, y=pa.pos, color=contaminant)) +
  geom_point() +
  xlab("Prevalence (blanks)") +
  ylab("Prevalence (samples)") +
  ggtitle("Threshold = 0.5") +
  geom_abline(slope = 1, intercept = 0)
```

I am not happy with how this identified contaminants. There are some that are still present in a very large number of samples and were only found in one contaminant. I am going to try lowering the strict threshold of 0.5 to 0.1 to see if that keeps some of these samples.  

```{r R1 decontam 0.1, echo = FALSE, message = FALSE}
# run decontam
contamdf.prevalence <- isContaminant(WC1, method="prevalence", neg="is.neg", threshold=0.1)

# identify which taxa are contaminants
contams <- which(contamdf.prevalence$contaminant)

# list contaminants
tax <- as(tax_table(WC1), "matrix")

# make separate presence-absence phyloseq for controls and samples
ps.pa <- transform_sample_counts(WC1, function(abund) 1*(abund>0))
ps.pa.neg <- prune_samples(sample_data(ps.pa)$sample.type == "blank", ps.pa)
ps.pa.pos <- prune_samples(sample_data(ps.pa)$sample.type == "fecal", ps.pa)

# make dataframe for prevalence in controls and samples, plus whether it was identified as a contaminant
df.pa <- data.frame(pa.pos=taxa_sums(ps.pa.pos), pa.neg=taxa_sums(ps.pa.neg),
                    contaminant=contamdf.prevalence$contaminant)

# plot prevalence and contaminant identification
ggplot(data=df.pa, aes(x=pa.neg, y=pa.pos, color=contaminant)) +
  geom_point() +
  xlab("Prevalence (blanks)") +
  ylab("Prevalence (samples)") +
  ggtitle("Threshold = 0.1") +
  geom_abline(slope = 1, intercept = 0)
```

```{r R1 contam table 0.1, echo = FALSE, render = lemon_print}
# print number of contamninants and what they are
as.data.frame(table(contamdf.prevalence$contaminant))
as.data.frame(tax[contams,])
```

These make a bit more sense as contaminants. Now I just want to make sure that I am not removing too many samples by removing these contaminants. This table shows % of reads removed from each sample after removing contaminants. The first two samples are the two blanks.  

```{r R1 remove contaminants, echo = FALSE, message = FALSE}
# remove taxa from phyloseq
WC1.decontam <- prune_taxa(!contamdf.prevalence$contaminant, WC1)

# set up df with readcounts and % removed
readcounts <- data.frame(matrix(ncol = 0, nrow = 124))
readcounts$pre <- sample_sums(WC1)
readcounts$post <- sample_sums(WC1.decontam)
readcounts$percent.removed <- ((readcounts$pre - readcounts$post)/readcounts$pre) * 100

readcounts$percent.removed
```

```{r R1 print table, echo = FALSE, render = lemon_print}
# print otu table for contaminants
as.data.frame(otu_table(WC1)) %>% slice(contams)
```
None of the samples lost more than 15/16 percent of their reads so I am good with the percent removed. The ASV table shows that some of them have higher read counts than in the blanks, but they are only in a few samples, and the blanks had much lower reads overall so that would skew the number.  

6 taxa were identified as contaminants using a threshold of 0.1 (would be a good idea for me to articulate what this threshold means). I have removed these taxa from analysis.

### Run 2 decontam

First I will use the extraction blanks to identify contaminants. Using a threshold of 0.1 (given that was the best threshold for Run 1), these are the taxa identified as contaminants:  

```{r R2 decontam 0.1, echo = FALSE, message = FALSE}
# make new variable where blanks are "TRUE"
sample_data(WC2)$is.neg <- sample_data(WC2)$sample.type == "blank"

# run decontam
contamdf.prevalence <- isContaminant(WC2, method="prevalence", neg="is.neg", threshold=0.1)

# identify which taxa are contaminants
contams <- which(contamdf.prevalence$contaminant)

# list contaminants
tax <- as(tax_table(WC1), "matrix")
```

```{r R2 contam table 0.1, echo = FALSE, render = lemon_print}
# print number of contamninants and what they are
as.data.frame(table(contamdf.prevalence$contaminant))
as.data.frame(tax[contams,])
```

I will also make a plot that shows prevalence of taxa in the negatives and in true samples, to check how they are getting classified.  

```{r R2 decontam plot, echo = FALSE, message = FALSE}
# make separate presence-absence phyloseq for controls and samples
ps.pa <- transform_sample_counts(WC2, function(abund) 1*(abund>0))
ps.pa.neg <- prune_samples(sample_data(ps.pa)$sample.type == "blank", ps.pa)
ps.pa.pos <- prune_samples(sample_data(ps.pa)$sample.type != "blank", ps.pa)

# make dataframe for prevalence in controls and samples, plus whether it was identified as a contaminant
df.pa <- data.frame(pa.pos=taxa_sums(ps.pa.pos), pa.neg=taxa_sums(ps.pa.neg),
                    contaminant=contamdf.prevalence$contaminant)

# plot prevalence and contaminant identification
ggplot(data=df.pa, aes(x=pa.neg, y=pa.pos, color=contaminant)) +
  geom_point() +
  xlab("Prevalence (blanks)") +
  ylab("Prevalence (samples)") +
  ggtitle("Threshold = 0.1") +
  geom_abline(slope = 1, intercept = 0)
```

This looks fine to me. Again, I want to make sure that I am not removing too many samples by removing these contaminants. This table shows % of reads removed from each sample after removing contaminants. The first three samples are the blanks.  

```{r R2 remove contaminants, echo = FALSE, message = FALSE}
# remove taxa from phyloseq
WC2.decontam <- prune_taxa(!contamdf.prevalence$contaminant, WC2)

# set up df with readcounts and % removed
readcounts <- data.frame(matrix(ncol = 0, nrow = 60))
readcounts$pre <- sample_sums(WC2)
readcounts$post <- sample_sums(WC2.decontam)
readcounts$percent.removed <- ((readcounts$pre - readcounts$post)/readcounts$pre) * 100

readcounts$percent.removed
```

```{r R2 print table, echo = FALSE, render = lemon_print}
# print otu table for contaminants
as.data.frame(otu_table(WC2)) %>% slice(contams)
```
None of the samples lost more than 7/8 percent of their reads so I am good with the percent removed.
In the end, 38 taxa were identified and removed using a threshold of 0.1 (would be a good idea for me to articulate what this threshold means). I have removed these taxa from analysis.

### Recombine groups and remove blanks

Before proceeding to analysis, I now need to recombine the R1 and R2 groups into Fecal and Tracheal phyloseqs. There were two fecal samples (donor feces and inoculum) that were included in R2 that for analysis I need to group with R1.  

During this step I will also remove the blanks.

```{r split R2, message = FALSE}
# isolate fecal from R2
WC2F <- subset_samples(WC2, sample.type == "fecal")

# isolate tracheal from R2
WCT <- subset_samples(WC2, sample.type == "tracheal")

# recombine R2 fecal with R1, and remove blanks
WCF <- merge_phyloseq(WC1, WC2F)
WCF <- subset_samples(WCF, sample.type == "fecal")
```

There are now two phyloseq objects: *WCF*, which contains the fecal samples (124 samples) and *WCT*, which contains the tracheal samples (55 samples). These will be used for further analysis.

## Rarefy 16S data

Now that I have the samples separated by fecal/tracheal, I can rarefy the respective groups for subsequent analysis.

### Rarefy fecal

First, I will look at read counts.

```{r F read counts, echo = FALSE, render = lemon_print}
# print read counts ordered from lowest to highest
samples <- as.data.frame(sample_sums(WCF))
arrange(samples, sample_sums(WCF))
```

Based on the read counts, I will rarefy at 16,378 which removes four samples: F386-14-2, F361-30-2, F391-50-2, and F383-30-2.

```{r F rarefy, include=FALSE, message = FALSE}
WCF.rare <- rarefy_even_depth(WCF, sample.size=16378, rngseed=14, verbose=TRUE) # rarefy
```

### Rarefy tracheal

```{r T read counts, echo = FALSE, render = lemon_print}
# print read counts ordered from lowest to highest
samples <- as.data.frame(sample_sums(WCT))
arrange(samples, sample_sums(WCT))
```

Based on the read counts, for now I will rarefy at 1073 which excludes four samples. In future, I would like to figure out a rarefaction curve for this data to see whether that is appropriate. Overall many of the reads have much lower counts than the fecal samples, but this is expected.

```{r T rarefy, include=FALSE, message = FALSE}
WCT.rare <- rarefy_even_depth(WCT, sample.size=1073, rngseed=14, verbose=TRUE) # rarefy
```

# Fecal 16S analysis

This will use the *WCF.rare* phyloseq object, except for differential abundance analyses which will use *WCF*.

## Alpha Diversity

After rarefying, I can look at the initial alpha diversity plots to look for differences. DI = Donor/Inoculum.

```{r F alpha all, echo = FALSE}
plot_richness(WCF.rare, x="group")
plot_richness(WCF.rare, x="sex")
plot_richness(WCF.rare, x="year", color = "group")
plot_richness(WCF.rare, x="day", color="group")
```


Some trends I see: PR might have slightly higher diversity than other groups (see Simpson especially), and the AR-FMT points seem to be a bit lower. Something is definitely going on with day 14. Also, the Donor/Inoculum definitely stick together which is great to see since the inoculum is derived from that fecal material. Generally the values seem to stay with what is typical overall.  

Digging into differences between groups, I can plot diversity per group over time. I removed day 21 for these plots because only one group has any data points at day 21.

```{r F alpha dataframe setup, include = FALSE, message = FALSE}
# save alpha diversity calculations as alpha dataframe
alpha <- microbiome::alpha(WCF.rare)

# save phyloseq metadata as separate object, and add shannon and observed richness
metaF.rare <- meta(WCF.rare) %>%
  mutate(shannon = alpha$diversity_shannon) %>%
  mutate(observed = alpha$observed)
```

Shannon diversity plot:  
```{r F shannon over time, echo = FALSE}
metaF.rare %>%
  filter(day != "21") %>% # remove day 21 samples because only AR had it
  ggplot(aes(x=day, y=shannon, group = group)) + sc +
  geom_point(aes(color = group), position = position_dodge(width = 0.1), shape = 1, size = 3) +
  stat_summary(fun=mean, geom = "line", linewidth = 1, aes(color = group)) +
  stat_summary(fun.data = data.summary, geom = "errorbar", linewidth = 1, width = 0.5, na.rm = TRUE, aes(color = group)) +
  stat_summary(fun=mean, geom = "point", size = 3, aes(color = group)) +
  theme_classic(base_size = 15) +
  ylab("shannon diversity")
```

```{r F richness over time, echo = FALSE}
metaF.rare %>%
  filter(day != "21") %>%
  ggplot(aes(x=day, y=observed, group = group)) + sc +
  geom_point(aes(color = group), position = position_dodge(width = 0.1), shape = 1, size = 3) +
  stat_summary(fun=mean, geom = "line", linewidth = 1, aes(color = group)) +
  stat_summary(fun.data = data.summary, geom = "errorbar", linewidth = 1, width = 0.5, na.rm = TRUE, aes(color = group)) +
  stat_summary(fun=mean, geom = "point", size = 3, aes(color = group)) +
  theme_classic(base_size = 15) +
  ylab("observed richness")
```

Seems like potentially, richness converges over time between the three groups to an intermediate "adult" richness. 

### Linear model (Shannon)
*INCLUDE HERE*


## Beta Diversity

### Initial plots

I will do some initial beta diversity plots to look for clustering patterns. I will do both Bray-Curtis and Jaccard.

```{r F beta div setup, include = FALSE}
# bray curtis
BrayDistF <- distance(WCF.rare, method = "bray") # create bray dist matrix
BrayOrdF <- ordinate(WCF.rare, "NMDS", distance = BrayDistF) # make NMDS ordination
BrayPlotF <- plot_ordination(WCF.rare, BrayOrdF, justDF = TRUE) # make base NMDS dataframe for plotting

# jaccard
JaccDistF <- distance(WCF.rare, method = "jaccard") # create jaccard matrix
JaccOrdF <- ordinate(WCF.rare, "NMDS", distance = JaccDistF) # make NMDS ordination
JaccPlotF <- plot_ordination(WCF.rare, JaccOrdF, justDF = TRUE) # make base NMDS dataframe for plotting
```

Plots colored by group:  
```{r F by group, echo = FALSE}
# bray curtis
BrayPlotF %>%
  ggplot(aes(x=NMDS1, y=NMDS2)) + sc +
  geom_point(size=4, aes(color = group)) +
  labs(shape = "day")

# jaccard
JaccPlotF %>%
  ggplot(aes(x=NMDS1, y=NMDS2)) + sc +
  geom_point(size=4, aes(color = group)) +
  labs(shape = "day")
```

Adding day as shape:  
```{r F by group and day, echo = FALSE}
# bray curtis
BrayPlotF %>%
  ggplot(aes(x=NMDS1, y=NMDS2)) + sc +
  geom_point(size=4, aes(color = group, shape = day)) +
  labs(shape = "day")

# jaccard
JaccPlotF %>%
  ggplot(aes(x=NMDS1, y=NMDS2)) + sc +
  geom_point(size=4, aes(color = group, shape = day)) +
  labs(shape = "day")
```

Two main initial observations: First, relative abundance does not seem to be driving the differences between groups at all since bray-curtis and jaccard plots look pretty much identical. Future analyses that focus on presence/absence data might be more interesting than relative abundance data. Second, there seems to be some element of the beta diversity converging over time (I think). Still not sure what a good way to test this would be. *Maybe average distance to centroid over time?*

### Venn diagram of shared taxa

Unweighted and weighted: 

```{r F venn shared taxa, message = FALSE, echo = FALSE}
ps_venn(WCF, "group", quantities = list(type = c("percent", "counts"), font = 2), fill = c("red", "blue", "green"))
ps_venn(WCF, "group", quantities = list(type = c("percent"), font = 2), fill = c("red", "blue", "green"), weight = TRUE, relative = TRUE)
```

# Tracheal 16S analysis

## Alpha Diversity

After rarefying, I can look at the initial alpha diversity plots to look for differences. DI = Donor/Inoculum.

```{r T alpha all, echo = FALSE}
plot_richness(WCT.rare, x="group")
plot_richness(WCT.rare, x="sex")
plot_richness(WCT.rare, x = "swab.type")
plot_richness(WCT.rare, x="year", color = "group")
plot_richness(WCT.rare, x="day", color="group")
```

I suspect that here we might be seeing a similar pattern to what was happening with the fecal richness (divergence at first with convergence over time). It was nice to see no obvious patterns according to swab type affecting richness.

Digging into differences between groups, I can plot diversity per group over time.

```{r T alpha dataframe setup, include = FALSE, message = FALSE}
# save alpha diversity calculations as alpha dataframe
alpha <- microbiome::alpha(WCT.rare)

# save phyloseq metadata as separate object, and add shannon and observed richness
metaT.rare <- meta(WCT.rare) %>%
  mutate(shannon = alpha$diversity_shannon) %>%
  mutate(observed = alpha$observed)
```

Shannon diversity plot:  
```{r T shannon over time, echo = FALSE}
metaT.rare %>%
  filter(day != "21") %>% # remove day 21 samples because only AR had it
  ggplot(aes(x=day, y=shannon, group = group)) + sc +
  geom_point(aes(color = group), position = position_dodge(width = 0.1), shape = 1, size = 3) +
  stat_summary(fun=mean, geom = "line", linewidth = 1, aes(color = group)) +
  stat_summary(fun.data = data.summary, geom = "errorbar", linewidth = 1, width = 0.5, na.rm = TRUE, aes(color = group)) +
  stat_summary(fun=mean, geom = "point", size = 3, aes(color = group)) +
  theme_classic(base_size = 15) +
  ylab("shannon diversity")
```

```{r T richness over time, echo = FALSE}
metaT.rare %>%
  filter(day != "21") %>%
  ggplot(aes(x=day, y=observed, group = group)) + sc +
  geom_point(aes(color = group), position = position_dodge(width = 0.1), shape = 1, size = 3) +
  stat_summary(fun=mean, geom = "line", linewidth = 1, aes(color = group)) +
  stat_summary(fun.data = data.summary, geom = "errorbar", linewidth = 1, width = 0.5, na.rm = TRUE, aes(color = group)) +
  stat_summary(fun=mean, geom = "point", size = 3, aes(color = group)) +
  theme_classic(base_size = 15) +
  ylab("observed richness")
```

Contrary to the fecal richness, with the tracheal samples PR seems to separate more from AR/AR-FMT at the D70 timepoint. There is much more overlap between the groups at D14.

### Linear model (Shannon)
*INCLUDE HERE*

## Beta Diversity

### Initial plots

I will do some initial beta diversity plots to look for clustering patterns. I will do both Bray-Curtis and Jaccard.

```{r T beta div setup, include = FALSE}
# bray curtis
BrayDistT <- distance(WCT.rare, method = "bray") # create bray dist matrix
BrayOrdT <- ordinate(WCT.rare, "NMDS", distance = BrayDistT) # make NMDS ordination
BrayPlotT <- plot_ordination(WCT.rare, BrayOrdT, justDF = TRUE) # make base NMDS dataframe for plotting

# jaccard
JaccDistT <- distance(WCT.rare, method = "jaccard") # create jaccard matrix
JaccOrdT <- ordinate(WCT.rare, "NMDS", distance = JaccDistT) # make NMDS ordination
JaccPlotT <- plot_ordination(WCT.rare, JaccOrdT, justDF = TRUE) # make base NMDS dataframe for plotting
```

Plots colored by group:  
```{r T by group, echo = FALSE}
# bray curtis
BrayPlotT %>%
  ggplot(aes(x=NMDS1, y=NMDS2)) + sc +
  geom_point(size=4, aes(color = group)) +
  labs(shape = "day")

# jaccard
JaccPlotT %>%
  ggplot(aes(x=NMDS1, y=NMDS2)) + sc +
  geom_point(size=4, aes(color = group)) +
  labs(shape = "day")
```

Adding day as shape:  
```{r T by group and day, echo = FALSE}
# bray curtis
BrayPlotT %>%
  ggplot(aes(x=NMDS1, y=NMDS2)) + sc +
  geom_point(size=4, aes(color = group, shape = day)) +
  labs(shape = "day")

# jaccard
JaccPlotT %>%
  ggplot(aes(x=NMDS1, y=NMDS2)) + sc +
  geom_point(size=4, aes(color = group, shape = day)) +
  labs(shape = "day")
```

This is a similar clustering pattern to what we saw with the fecal samples. Again, there are quite similar patterns between bray curtis and jaccard. There still seems to be an element of clustering over time!

### Venn diagram of shared taxa

Unweighted and weighted: 

```{r T venn shared taxa, message = FALSE, echo = FALSE}
ps_venn(WCT, "group", quantities = list(type = c("percent", "counts"), font = 2), fill = c("red", "blue", "green"))
ps_venn(WCT, "group", quantities = list(type = c("percent"), font = 2), fill = c("red", "blue", "green"), weight = TRUE, relative = TRUE)
```

